src/validation/joi/joi.shared.schemas.ts

```diff
@@ -29,6 +29,10 @@ export const dateIntervalStringSchema = stringSchema.regex(DATE_INTERVAL_REGEX).
   'string.pattern.base': `must be a DateInterval string`,
 })
 
+export const DATE_TIME_STRING_REGEX =
+  /^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}(?::\d{2})?(?:Z|[+-]\d{2}:\d{2})?$/
+export const dateTimeStringSchema = stringSchema.regex(DATE_TIME_STRING_REGEX)
```

Comment:
p3: Bonus points: somewhere around this file there must be an example of how to give a custom error message in case validation fails. Without a custom message - it'll throw somewhat-ugly error with "X does not match regex #@$$%!@#@!$@#$". With a custom message we can say "X must be a DateTimeString" (see e.g a custom message for DateInterval).